### DISCUSSION TITLE
Freeform Triggers Two Events on Entry Save

### DISCUSSION THREAD

**Discussion by kevinamezaga** [solspace/craft-freeform#208] [Questions & Answers] [answered]
Hi there,

I use Freeform to edit an element to add a user + change a lightswitch on an entry. When the entry saves, it sends an email to the user that was just added and the author of that entry.

When the entry is saved via the control panel, it sends on email to each user. When the entry is saved/changed from the front end via Freeform, two emails are sent out.

I'm not sure why this is happening and it doesn't happen on another site with the same use-case and almost verbatim code. Any ideas?

**Comment by kjmartens** [comment, marked as answer]
Hi @kevinamezaga,

I'm curious how you're generating these notifications to begin with. I don't believe this is a feature built into Craft? :)

**Reply by kevinamezaga** [reply]
Hi Kelsey,

I found out it's an issue with the module I have to send out notifications. For some reason, Craft keeps marking one specific field as dirty even though it's not >:-(

in  beforeSave

![Screen Shot 2021-10-07 at 10 41 04](https://user-images.githubusercontent.com/25387497/136407817-1dc8c6f1-b345-440c-b6c7-bafa083e1c79.png)


in afterSave 

![Screen Shot 2021-10-07 at 10 36 51](https://user-images.githubusercontent.com/25387497/136407803-cc90ad7f-0a7c-466a-adfe-a236d391d064.png)

**Reply by kevinamezaga** [reply]
@kjmartens: found out why this was happening:

I'm using a preparser field to generate an amount of relations to the element. By default, the preparser field saves after the element saves—which would be fine, but what it does is then sets all the fields as dirty, bypassing all of my plugin's logic. I set the field to parse before save which fixed up everything.

Not Freeform's fault! I'm still at a loss for why it only happened when saving an element with Freeform though, but I think it's for another reason and not Freeform.

**Reply by kjmartens** [reply]
Thanks for following up! 🙂
